src/tests/java/htsjdk/samtools/util/zip/IntelDeflaterTest.java

```diff
@@ -0,0 +1,125 @@
+    }
+
+    private static final Log log = Log.getInstance(IntelDeflaterTest.class);
+
+    public static void main(String[] args) throws IOException {
+        if (args.length < 4) {
+            System.out.println("Usage: " + IntelDeflaterTest.class.getCanonicalName() + " inFile eagerDecode [outFile]");
+            System.exit(1);
+        }
+        final File inputFile = new File(args[0]);
+        final boolean eagerDecode = Boolean.parseBoolean(args[1]); //useful to test (realistic) scenarios in which every record is always fully decoded.
+        final File outputFile = new File(args[2]);
+
+	final int compressionLevel = Integer.parseInt(args[3]);
+
+
+        final long start = System.currentTimeMillis();
+
+	if (!DeflaterFactory.usingIntelDeflater()) {
+	    System.out.println("Running without IntelDeflater. Please specify the path to enable IntelDeflater, exiting");
+	    System.exit(1);
+	}
+
+        log.info("Start with args:" + Arrays.toString(args));
+        printConfigurationInfo();
+
+        SamReaderFactory readerFactory = SamReaderFactory.makeDefault().validationStringency(ValidationStringency.SILENT);
+        if (eagerDecode) {
+            readerFactory = readerFactory.enable(SamReaderFactory.Option.EAGERLY_DECODE);
+        }
+
+
+        try (final SamReader reader = readerFactory.open(inputFile)) {
+            final SAMFileHeader header = reader.getFileHeader();
+            final SAMFileWriter writer = new SAMFileWriterFactory().makeBAMWriter(header, true, outputFile, compressionLevel);
+	    final ProgressLogger pl = new ProgressLogger(log, 1000000);
+	    for (final SAMRecord record : reader) {
+		if (writer != null) {
+		    writer.addAlignment(record);
+		}
+		pl.record(record);
+	    }
+	    writer.close();
+	    log.info("BAM File Written using IntelDeflater");
+	    log.info("Read the outputted BAM now to ensure correctness");
+
+	    try (final SamReader outputReader = readerFactory.open(outputFile)) {
+		    for (final SAMRecord record : outputReader) {
+			pl.record(record);
+		    }
+		} catch (Exception e) {
+		log.error("Error reading record written with the IntelDeflater library");
+		System.exit(1);
+	    }
+        }
+	
+        final long end = System.currentTimeMillis();
+        log.info(String.format("Done. Elapsed time %.3f seconds", (end - start) / 1000.0));
+    }
+
+    private static void printConfigurationInfo() throws IOException {
+        log.info("Executing as " +
+                System.getProperty("user.name") + '@' + InetAddress.getLocalHost().getHostName() +
+                " on " + System.getProperty("os.name") + ' ' + System.getProperty("os.version") +
+                ' ' + System.getProperty("os.arch") + "; " + System.getProperty("java.vm.name") +
+                ' ' + System.getProperty("java.runtime.version") +
+                ' ' + (DeflaterFactory.usingIntelDeflater() ? "IntelDeflater" : "JdkDeflater"));
+        log.info("CREATE_INDEX:" + Defaults.CREATE_INDEX +
```

Comment:
this is misleading. should probably use the values from the writer. I just tried is with compression level 1 and it said 5...